sweep/sweeper.go

```diff
@@ -948,6 +981,135 @@ func (s *UtxoSweeper) handlePendingSweepsReq(req *pendingSweepsReq) []*PendingIn
 	return pendingInputs
 }
 
+// BumpFee allows bumping the fee of an input according to the provided fee
+// preference. Depending on the UtxoSweeper's knowledge of the input, either a
+// replacement transaction (RBF) will be created to allow the new fee rate to
+// propagate throughout the network, or a child transaction will be created to
+// cover its parents fees (CPFP).
+//
+// NOTE: This currently doesn't do any fee rate validation to ensure that a bump
+// is actually successful. The responsibility of doing so should be handled by
+// the caller.
+func (s *UtxoSweeper) BumpFee(input input.Input,
+	feePreference FeePreference) (chan Result, error) {
+
+	if input == nil || input.OutPoint() == nil {
+		return nil, errors.New("nil input received")
+	}
+
+	// Ensure the client provided a sane fee preference.
+	if _, err := s.feeRateForPreference(feePreference); err != nil {
+		return nil, err
+	}
+
+	responseChan := make(chan *bumpFeeResp, 1)
+	select {
+	case s.bumpFeeReqs <- &bumpFeeReq{
+		input:         input,
+		feePreference: feePreference,
+		responseChan:  responseChan,
+	}:
+	case <-s.quit:
+		return nil, ErrSweeperShuttingDown
+	}
+
+	select {
+	case response := <-responseChan:
+		return response.resultChan, response.err
+	case <-s.quit:
+		return nil, ErrSweeperShuttingDown
+	}
+}
+
+// handleBumpFee handles an incoming request to bump the fee rate of a given
+// input. Depending on the UtxoSweeper's knowledge of the input, either a
+// replacement transaction (RBF) will be created to allow the new fee rate to
+// propagate throughout the network, or a child transaction will be created to
+// cover its parents fees (CPFP).
+func (s *UtxoSweeper) handleBumpFeeReq(req *bumpFeeReq,
+	bestHeight int32) (chan Result, error) {
+
+	// If the UtxoSweeper is already trying to sweep this input, then we can
+	// simply just increase its fee rate. This will allow the input to be
+	// batched with others which also have a similar fee rate, creating a
+	// higher fee rate transaction that replaces the original input's
+	// sweeping transaction.
+	if pendingInput, ok := s.pendingInputs[*req.input.OutPoint()]; ok {
+		return s.handleRBFBump(req, pendingInput, bestHeight)
```

Comment:
It feels like the `Input` struct isn't the best fit for rbf bump. Maybe `RBFBump` should be exposed on the sweeper separately. In the `rpcserver` we already determined whether it was rbf or cpfp?

Reply:
Agreed. Having one call to bump fees is only necessary at the RPC level, we can abstract things better within the sweeper.

Reply:
Done. `UtxoSweeper.BumpFee` is now only meant to be used for inputs already within the sweeper.  To handle CPFPs, we can just go through the usual `UtxoSweeper.SweepInput` path.